packages/dev/loaders/src/glTF/2.0/Extensions/EXT_lights_area.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type { Nullable } from "core/types";
+import { Vector3, Quaternion } from "core/Maths/math.vector";
+import { Color3 } from "core/Maths/math.color";
+import { Light } from "core/Lights/light";
+import { RectAreaLight } from "core/Lights/rectAreaLight";
+import type { TransformNode } from "core/Meshes/transformNode";
+import { TransformNode as BabylonTransformNode } from "core/Meshes/transformNode";
+
+import type { IEXTLightsArea_LightReference } from "babylonjs-gltf2interface";
+import { EXTLightsArea_LightShape } from "babylonjs-gltf2interface";
+import type { INode, IEXTLightsArea_Light } from "../glTFLoaderInterfaces";
+import type { IGLTFLoaderExtension } from "../glTFLoaderExtension";
+import { GLTFLoader, ArrayItem } from "../glTFLoader";
+import { registerGLTFExtension, unregisterGLTFExtension } from "../glTFLoaderExtensionRegistry";
+
+const NAME = "EXT_lights_area";
+
+declare module "../../glTFFileLoader" {
+    // eslint-disable-next-line jsdoc/require-jsdoc, @typescript-eslint/naming-convention
+    export interface GLTFLoaderExtensionOptions {
+        /**
+         * Defines options for the EXT_lights_area extension.
+         */
+        // NOTE: Don't use NAME here as it will break the UMD type declarations.
+        ["EXT_lights_area"]: {};
+    }
+}
+
+/**
+ * [Specification](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Vendor/EXT_lights_area/README.md)

```

Comment:
Definitely should be marked experimental as the spec hasn't been merged yet. The link also doesn't work.